src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -239,20 +239,53 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         }
     }
 
+    private isTaggableAssetType = (asset: IAsset): boolean => {
+        return asset.type === AssetType.Image || asset.type === AssetType.VideoFrame;
+    }
+
     /**
      * Raised when the selected asset has been changed.
      * This can either be a parent or child asset
      */
     private onAssetMetadataChanged = async (assetMetadata: IAssetMetadata): Promise<void> => {
-        assetMetadata.asset.state = assetMetadata.regions.length > 0 ? AssetState.Tagged : AssetState.Visited;
+        const parentAsset = { ...(assetMetadata.asset.parent || assetMetadata.asset) };
+
+        if (this.isTaggableAssetType(assetMetadata.asset)) {
+            assetMetadata.asset.state = assetMetadata.regions.length > 0 ? AssetState.Tagged : AssetState.Visited;
+        } else if (assetMetadata.asset.state === AssetState.NotVisited) {
+            assetMetadata.asset.state = AssetState.Visited;
+        }
+
+        // Update parent asset if not already in the "Tagged" state
+        if (parentAsset.id === assetMetadata.asset.id) {
```

Comment:
this confused me.  we get `parentAsset` from `assetMetadata` so wouldn't id be the same?

Reply:
Renamed to `rootAsset` to make this more clear.